src-plugins/itkFilters/itkFiltersInvertProcess_p.h

```diff
@@ -41,9 +42,19 @@ class itkFiltersInvertProcessPrivate : public itkFiltersProcessBasePrivate
         typedef itk::Image< PixelType, 3 > ImageType;
         typedef itk::InvertIntensityImageFilter< ImageType, ImageType >  InvertFilterType;
         typename InvertFilterType::Pointer invertFilter = InvertFilterType::New();
-    
+
+        // compute maximum intensity of image
+        typedef itk::MinimumMaximumImageCalculator< ImageType > MinMaxFilterType;
+        typename MinMaxFilterType::Pointer maxFilter = MinMaxFilterType::New();
+
+        maxFilter->SetImage( dynamic_cast<ImageType *> ( ( itk::Object* ) ( input->data() ) ) );
+        maxFilter->Compute();
+        PixelType maximum = maxFilter->GetMaximum();
+        PixelType minimum = maxFilter->GetMinimum();
+
         invertFilter->SetInput ( dynamic_cast<ImageType *> ( ( itk::Object* ) ( input->data() ) ) );
-        
+        invertFilter->SetMaximum(maximum + minimum);
```

Comment:
this puzzled me as well so I did the computation myself: the formula corresponding to what we want to do is inverting the centered values and then adding the mean again, which means if I1 is the original image and I2 the filtered image, we want the output to be:

I2 = - (I1 - (min+max)/2) + (min+max)/2 = - I1 + (min + max) = (min + max) - I1

if you read the documentation, it says "InvertIntensityImageFilter inverts intensity of pixels by subtracting pixel value to a maximum value", which is the "maximum" parameter.

so effectively we want to subtract I1 to a maximum value of (min + max), this is why you have to put (min + max) as maximum value of the filter, because you don't give the maximum value of the final image but the value you want the image to be subtracted to.

hope I am clear enough, but in any case the code makes sense.